File: GeeksForGeeks/Leaders In an Array/Leaders in an Array.py
# Python function to print leaders in array 
def printLeaders(l, size): 
    m=[]
    max_element = l[size-1]  
    m.append(max_element)     
    for i in range(size-2, -1, -1):         
        if max_element <= l[i]: 
            m.append(l[i])
            max_element = l[i] 
            
    for k in range(len(m)-1,-1,-1):  #get the elements according to their first occurence in the arr
        print(m[k],end=" ")
          
# Driver function 
arr = [16, 17, 4, 3, 5, 2] 
printLeaders(arr, len(arr))

File: GeeksForGeeks/Leaders In an Array/leaders_in_an_array.java
// Program to find all leaders in an array

// An element is leader if it is greater than all the elements to its right side of an array.
// And the rightmost element is always a leader.

void printLeaders(int arr[], int n)
{
    int maximum =  arr[n-1];
    int i;

    // Rightmost element is always leader
    printf("%d ", maximum);

    for(i = n-2; i >= 0; i--)
    {
        if(maximum < arr[i])
        {
           printf("%d ", arr[i]);
           maximum = arr[i];
        }
    }   
}